src/Workspaces/Core/Portable/Utilities/Contract.cs

```diff
@@ -11,7 +14,7 @@ internal static class Contract
         /// Throws a non-accessible exception if the provided value is null.  This method executes in
         /// all builds
         /// </summary>
-        public static void ThrowIfNull<T>(T value, string message = null) where T : class
+        public static void ThrowIfNull<T>([NotNull] T value, string? message = null) where T : class?
```

Comment:
Just to make sure I understand, we're asserting T can be a nullable ref type, but can't be passed in as null? Does that make sense here, instead of `DoesNotReturnIf(value: null)` ?

Reply:
DoesNotReturnIf only takes Boolean-valued return values so I can't write that (although yes, it'd be more intuitive). The value of this annotation is saying that yes, we can pass in a null, but after that call the value can be assumed to be non-null. Tagging @333fred if there's a better way to annotate this.

Reply:
This is the correct way to annotate.

Reply:
`ThrowIfNull<T>([NotNull] T)` looks so weird :(